text/0162-refactoring-support.md

```diff
@@ -0,0 +1,702 @@
+  choose to only deploy stack A, and a resource was moved from stack A to stack
+  B, the refactor will involve both stacks. But if there was a rename in, let's
+  say, stack C, it will not be refactored. The same set of filters is available
+  for the `refactor` command.
+- To perform refactoring, the CLI needs new permissions in the bootstrap stack.
+  Before using this feature, run `cdk bootstrap` for every target environment,
+  to add these new permissions.
+- A given CDK application can have multiple stacks, for different environments.
+  In that case, the CLI will group the stacks by environment and perform the
+  refactoring separately in each one. So, although you can move resources
+  between stacks, both stacks involved in the move must be in the same
+  environment. Trying to move resources across environments will result in an
+  error.
+
+### Explicit mapping
+
+The previous section describes an interaction between you and the CDK CLI, in
+which you are presented with a candidate for refactoring, and you decide whether
+you want it to happen. In a CI/CD pipeline, however, there is no human to make
+this decision. In this case, you can use explicit mappings to define your own
+refactors.
+
+Here is how it works: at development time, you made a change that the CLI
+detected as a refactor. Since you want that refactor to be propagated to other
+environments, you export the mapping file, with the command
+`cdk refactor --record-resource-mapping=file.json`.
+
+There are at least two possible paths from here, depending on your
+circumstances:
+
+1. You send the exported file to an operations team, who will review it. If
+   approved, they manually run the command
+   `cdk refactor --resource-mapping=file.json` on every protected environment in
+   advance (i.e., before your changes get deployed to those environments). When
+   you import a mapping, the CLI won't try to detect refactors.
+2. The `--resource-mapping` option is also available for the `deploy` command.
+   So you can commit the mapping file to version control, and configure your
+   pipeline to use it on every deployment. This is a more convenient option, as
+   it requires less coordination between different roles. Every time a refactor
+   is applied, a record of it is stored in the environment. This is to prevent
+   the same refactor from being applied multiple times.
+
+You can also use explicit mappings to define your own refactors, when the CLI
+didn't detect them automatically. This may happen a resource is moved and
+modified at the same time, for example. You may also be in a situation in which
+the CLI detected a refactor, but you actually intend to replace it. To
+accomplish this, you can remove the unwanted mapping from the mapping file.
+
+### Settings
+
+You have a few settings available to control the behavior of the refactoring
+feature.
+
+For both `deploy` and `refactor`:
+
+- `--record-resource-mapping=<FILE>`: writes the mapping to a file. The file can
```

Comment:
I am unclear if when I use this it only writes the mapping file out OR it does the remapping and writes a file out (to apply elsewhere)?